src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ITextViewWindowInProcessExtensions.cs

```diff
@@ -0,0 +1,252 @@
+
+            var view = await textViewWindow.TestServices.Editor.GetActiveTextViewAsync(cancellationToken);
+            var broker = await textViewWindow.TestServices.Shell.GetComponentModelServiceAsync<IAsyncQuickInfoBroker>(cancellationToken);
+
+            var session = broker.GetSession(view);
+
+            // GetSession will not return null if preceded by a call to InvokeQuickInfo
+            Contract.ThrowIfNull(session);
+
+            while (session.State != QuickInfoSessionState.Visible)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await Task.Delay(50, cancellationToken).ConfigureAwait(true);
+            }
+
+            return QuickInfoToStringConverter.GetStringFromBulkContent(session.Content);
+        }
+
+        public static async Task ShowLightBulbAsync(this ITextViewWindowInProcess textViewWindow, CancellationToken cancellationToken)
+        {
+            await textViewWindow.TestServices.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var shell = await textViewWindow.TestServices.Shell.GetRequiredGlobalServiceAsync<SVsUIShell, IVsUIShell>(cancellationToken);
+            var cmdGroup = typeof(VSConstants.VSStd14CmdID).GUID;
+            var cmdExecOpt = OLECMDEXECOPT.OLECMDEXECOPT_DONTPROMPTUSER;
+
+            var cmdID = VSConstants.VSStd14CmdID.ShowQuickFixes;
+            object? obj = null;
+            shell.PostExecCommand(cmdGroup, (uint)cmdID, (uint)cmdExecOpt, ref obj);
+
+            var view = await textViewWindow.GetActiveTextViewAsync(cancellationToken);
+            var broker = await textViewWindow.TestServices.Shell.GetComponentModelServiceAsync<ILightBulbBroker>(cancellationToken);
+            await LightBulbHelper.WaitForLightBulbSessionAsync(textViewWindow.TestServices, broker, view, cancellationToken);
+        }
+
+        public static async Task InvokeCompletionListAsync(this ITextViewWindowInProcess textViewWindow, CancellationToken cancellationToken)
+        {
+            await textViewWindow.TestServices.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            await textViewWindow.TestServices.Shell.ExecuteCommandAsync(WellKnownCommands.Edit.ListMembers, cancellationToken);
+            await textViewWindow.TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.CompletionSet, cancellationToken);
+        }
+
+        public static async Task<ImmutableArray<Completion>> GetCompletionItemsAsync(this ITextViewWindowInProcess textViewWindow, CancellationToken cancellationToken)
+        {
+            await textViewWindow.TestServices.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            await WaitForCompletionSetAsync(textViewWindow, cancellationToken);
+
+            // It's not known why WaitForCompletionSetAsync fails to stabilize calls to GetCompletionItemsAsync.
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
📝 Added a delay here as well. It looks like `ICompletionBroker` is introducing significant instability in the overall testing. We should work with the editor to stabilize these APIs.

Reply:
yea, we already switched to another API based on Kayle's suggestion, but it didn't help
https://github.com/dotnet/roslyn/pull/70578/files